src/main/java/se/sundsvall/paratransit/integration/casedata/mapper/CaseDataMapper.java

```diff
@@ -0,0 +1,150 @@
+package se.sundsvall.paratransit.integration.casedata.mapper;
+
+import static java.time.OffsetDateTime.now;
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+import static java.util.Collections.emptyList;
+import static java.util.Optional.ofNullable;
+import static se.sundsvall.paratransit.Constants.CASEDATA_KEY_DISPLAY_PHASE;
+import static se.sundsvall.paratransit.Constants.CASEDATA_KEY_PHASE_ACTION;
+import static se.sundsvall.paratransit.Constants.CASEDATA_KEY_PHASE_STATUS;
+
+import generated.se.sundsvall.casedata.Attachment;
+import generated.se.sundsvall.casedata.Decision;
+import generated.se.sundsvall.casedata.Decision.DecisionOutcomeEnum;
+import generated.se.sundsvall.casedata.Decision.DecisionTypeEnum;
+import generated.se.sundsvall.casedata.Errand;
+import generated.se.sundsvall.casedata.ExtraParameter;
+import generated.se.sundsvall.casedata.Law;
+import generated.se.sundsvall.casedata.MessageAttachment;
+import generated.se.sundsvall.casedata.MessageRequest;
+import generated.se.sundsvall.casedata.MessageRequest.DirectionEnum;
+import generated.se.sundsvall.casedata.PatchErrand;
+import generated.se.sundsvall.casedata.Stakeholder;
+import generated.se.sundsvall.casedata.Stakeholder.TypeEnum;
+import generated.se.sundsvall.casedata.Status;
+import generated.se.sundsvall.templating.RenderResponse;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class CaseDataMapper {
+
+	private CaseDataMapper() {}
+
+	public static PatchErrand toPatchErrand(final String externalCaseId, final String phase, final String phaseStatus, final String phaseAction, final String displayPhase, final List<ExtraParameter> extraParameters) {
+		final var patchErrand = toPatchErrand(externalCaseId, phase, phaseStatus, phaseAction, extraParameters);
+
+		var result = patchErrand.getExtraParameters().stream()
+			.filter(extraParameter -> !CASEDATA_KEY_DISPLAY_PHASE.equals(extraParameter.getKey()))
+			.collect(Collectors.toCollection(ArrayList::new));
+
+		Optional.ofNullable(displayPhase).ifPresentOrElse(display -> result.add(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(List.of(display))),
+			() -> result.add(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(emptyList())));
+
+		return patchErrand.extraParameters(result);
+	}
+
+	public static PatchErrand toPatchErrand(final String externalCaseId, final String phase, final String phaseStatus, final String phaseAction, final List<ExtraParameter> extraParameters) {
+		final var patchErrand = new PatchErrand()
+			.externalCaseId(externalCaseId)
+			.phase(phase);
+
+		var result = Optional.ofNullable(extraParameters).orElse(emptyList()).stream()
```

Comment:
Could use a clearer name perhaps? parameters or something? (Goes for row 40 too)